packages/ui/primitives/stepper/stepper-context.ts

```diff
@@ -0,0 +1,29 @@
+import { createContext, useContext } from 'react';
+
+type StepperContext = {
+  currentStep: number;
+  totalSteps: number;
+  isFirstStep: boolean;
+  isLastStep: boolean;
+  nextStep: () => void;
+  prevStep: () => void;
+};
+
+export const StepperProvider = createContext<StepperContext>({
+  currentStep: 1,
+  totalSteps: 1,
+  isFirstStep: true,
+  isLastStep: false,
+  nextStep: () => undefined,
+  prevStep: () => undefined,
+});
+
+export const useStepper = () => {
+  const context = useContext(StepperProvider);
+
+  if (!context) {
+    throw Error('useContext hook should be called inside the Stepper Provider');
+  }
+
+  return context;
+};
```

Comment:
If you're using this pattern it's best to make the default value null otherwise the `!context` won't get triggered since it'll use the value you provided as the `createContext` arg